### PR TITLE
Remove pytz in favour of Python's builtin ZoneInfo

### PR DESCRIPTION
**Pull request by jamesbeith**
Support for pytz is deprecated and will be removed in Django v5.

### CODE REVIEW

File: xocto/localtime.py [hunk 1]
```diff
@@ -68,16 +72,12 @@ def datetime(
...
 ) -> datetime_.datetime:
     """
     Return a datetime in the local timezone.
-
-    A boolean value is required for is_dst to unambiguously determine the appropriate dt during the
-    window when the hour goes back.
     """
     dt = datetime_.datetime(year, month, day, hour, minute, second, microsecond)
-    return timezone.make_aware(dt, is_dst=is_dst)
+    return timezone.make_aware(dt)
```
Comment (daisylb) [inline xocto/localtime.py:80]: Would it make sense to retain the `is_dst` parameter and convert it to the appropriate `fold` value on the `datetime` constructor?

Comment (jamesbeith) [reply xocto/localtime.py:80]: Unfortunately that's not quite straight forward as the two don’t map onto one another perfectly.

https://pytz-deprecation-shim.readthedocs.io/en/latest/migration.html#semantic-differences-between-is-dst-and-fold

**Review by codeinthehole — APPROVED** [review]
Looks good.

File: tests/test_localtime.py [hunk 2]
```diff
@@ -282,33 +290,23 @@ class TestDateTime:
...
-    # occurs twice).
-
     def test_dst_ambiguity(self):
-        with pytest.raises(pytz.AmbiguousTimeError):
-            localtime.datetime(2020, 10, 25, 1, 30)
-
-    def test_passing_dst_value_resolves_ambiguity(self):
-        dt_dst = localtime.datetime(2020, 10, 25, 1, 30, is_dst=True)
+        # 2020-10-25T01:30 is an ambiguous dt as its in the period when clocks go back (so it
+        # occurs twice).
```
[the remark this answers is not in the document]
Comment (jamesbeith) [reply tests/test_localtime.py]: Done in d733d9c99bfbb28e8661ccfa48a5c30ce1289779

File: xocto/localtime.py [hunk 3]
```diff
@@ -5,7 +5,11 @@
 import decimal
 from typing import Generator, Optional, Sequence, Tuple, Union
 
-import pytz
+try:
+    import zoneinfo
+except ImportError:
+    from backports import zoneinfo
```
[the remark this answers is not in the document]
Comment (jamesbeith) [reply xocto/localtime.py]: I've been able to remove the try..except block now as you kindly recently upgraded xocto to Python 3.9+ 👍🏼

File: xocto/localtime.py [hunk 4]
```diff
@@ -68,16 +72,12 @@ def datetime(
     minute: int = 0,
     second: int = 0,
     microsecond: int = 0,
-    is_dst: bool | None = None,
```
[the remark this answers is not in the document]
Comment (jamesbeith) [reply xocto/localtime.py:63]: Correct, we'll be bumping to v4.

File: tests/test_localtime.py [hunk 5]
```diff
@@ -282,33 +290,23 @@ class TestDateTime:
...
+        dt_dst = localtime.datetime(2020, 10, 25, 1, 30)
```
[the remark this answers is not in the document]
Comment (jamesbeith) [reply tests/test_localtime.py]: Fixed in d42c565427c0c389c877e2c8ce7d13834100a86b

File: xocto/localtime.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (pydanny) [reply xocto/localtime.py:63]: Thanks to both of you for being semver correct. I can't believe how pedantic I've gotten about that. 😉

**Review by pydanny — APPROVED** [review]
LGTM

**Pull request merged**